### ISSUE TITLE
Error: zrep_expire Internal Err caller did not hold fs lock (while 'scrub' is running on remote machine)

### ISSUE THREAD

**Issue by 3n4rch3** [bolthole/zrep#114]
I've done a lot of googling, and have looked closely at the suggested related items, but none *seem* to be quite the same issue.  Please have mercy if I missed the trail!

We are running a 'pull' zrep configuration, and everything has seemed to be running fine. However, after I set up email notifications for my cron jobs, I got spammed one day with emails declaring "zrep_expire Internal Err caller did not hold fs lock".

Long story short, it only happens when the remote (master?) machine is doing a scrub. As soon as the scrub is finished, zrep is able to once again expire snaps, and the next 'zrep refresh' cleans up the accumulated extra snaps.

Is there any known fix/workaround for this, or am I doomed to deleting spam while the weekly scrub runs?

**Comment by ppbrown** [comment]
workaround is; insert a check for “is scrub running?”  in your cron job?

**Comment by ppbrown** [comment]
lets be clear here: the behaviour is a bug in your ZFS implementation. So
the real solution is to complain to your driver writers.
Until then the possible workarounds are either the one I just suggested, or
to stop doing scrubs

**Comment by 3n4rch3** [comment]
Got it, thanks for clarifying.
